pkg/parser/Makefile

```diff
@@ -1,15 +1,17 @@
-.PHONY: all parser clean
-
-all: fmt parser generate
+.PHONY: all
+all: fmt parser
 
+.PHONY: test
 test: fmt parser
 	sh test.sh
 
-parser: parser.go hintparser.go
```

Comment:
Do we need to modify root/Makefile to check the lint?

https://github.com/pingcap/tidb/pull/54827/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R147